src/si/time.rs

```diff
@@ -44,3 +46,34 @@ quantity! {
         @year: 3.1536_E7; "a", "year", "years";
     }
 }
+
+impl<U, V> From<Time<U, V>> for Duration
+where
+    U: ::si::Units<V> + ?Sized,
+    V: ::num::Num + ::num::AsPrimitive<f64> + ::Conversion<V>,
+{
+    fn from(t: Time<U, V>) -> Duration {
+        let secs = t.value.as_().abs();
+        let nanos = (secs * 1e9) as u64 % 1e9 as u64;
```

Comment:
I'm worried about overflow here. It would be good to find a way to extract the nanoseconds without multiplication. Does converting the original value into nanoseconds and then extracting the portion needed for `Duration` work?

Reply:
Converting the original value to nanoseconds is a multiplication operation... there's no way around that part.  That said, taking off the integer part before multiplying can solve any potential overflow problems:  `let nanos = secs.fract() * 1e9;`.  It doesn't solve any of the base problems in your other point, however...